startups/lib/main_9.dart

```diff
@@ -0,0 +1,220 @@
+                  }),
+                ),
+              ),
+              Expanded(
+                child: Container(
+                  color: Theme.of(context).colorScheme.secondaryContainer,
+                  child: page,
+                ),
+              ),
+            ],
+          ),
+        );
+      },
+    );
+  }
+}
+
+class GeneratorPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    var appState = context.watch<MyAppState>();
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Expanded(
+          flex: 2,
+          child: ListView(
+            reverse: true,
+            children: [
+              for (var historicalPair in appState.history)
+                Center(
+                  child: Text(historicalPair.asLowerCase),
+                ),
+            ],
+          ),
+        ),
+        SizedBox(height: 16),
+        BigCard(appState.current),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: [
+            ElevatedButton.icon(
+              icon: Icon(appState.favorites.contains(appState.current)
+                  ? Icons.favorite
+                  : Icons.favorite_border),
+              label: Text('Like'),
+              onPressed: () {
+                appState.toggleCurrentFavorite();
+              },
+            ),
+            SizedBox(width: 8),
+            ElevatedButton.icon(
+              icon: Icon(Icons.refresh),
+              label: Text('Next'),
+              onPressed: () {
+                appState.generateNext();
+              },
+            ),
+            SizedBox(width: 28),
+          ],
+        ),
+        Spacer(),
+      ],
+    );
+  }
+}
+
+class BigCard extends StatelessWidget {
+  final WordPair pair;
+
+  BigCard(this.pair);
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    var style = theme.textTheme.displayMedium!.copyWith(
+      color: theme.colorScheme.onPrimary,
+    );
+
+    return Card(
+      margin: EdgeInsets.all(24),
+      color: theme.colorScheme.primary,
+      child: Padding(
+        padding: EdgeInsets.all(16),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Text(pair.first,
+                style: style.copyWith(fontWeight: FontWeight.w100)),
+            Text(
+              pair.second,
+              style: style.copyWith(fontWeight: FontWeight.bold),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class FavoritesPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    var appState = context.watch<MyAppState>();
+    var theme = Theme.of(context);
+
+    return Padding(
+      padding: EdgeInsets.all(16),
+      child: ListView(
+        children: [
+          for (final favorite in appState.favorites)
+            ListTile(
```

Comment:
Here, we're teaching `ListTile`. This whole screen is kind of meant to be a challenge to the student to make it their own. Some functionality, like removing favorites or showing something when the list is empty, is notably missing, and by the end of the codelab, motivated students should have no problem fixing these issues.